[release] Build vmtest binary for aarch64 and s390x too

chantra opened the pull request:
Build vmtest for x86_64, aarch64, and s390x.
As a side effect, x86_64 does not need a x86_64 anymore to be compiled.

Example release: https://github.com/chantra/danobi-vmtest/releases/tag/v1.0.0

danobi reviewed (approved):
>  As a side effect, x86_64 does not need a x86_64 anymore to be compiled.

Not sure I got what you mean by that, but lgtm here. Might be nice to pull the code into a file in `scripts/` but not a big deal

chantra commented:
> > As a side effect, x86_64 does not need a x86_64 anymore to be compiled.
> 
> Not sure I got what you mean by that, but lgtm here.

Because we install the $arch toolchain regardless of what the host architecture is.

ex: if host is arm64, we will still run:
```
sudo apt install -y gcc-${arch//_/-}-linux-gnu
rustup target add "${arch}-unknown-linux-gnu"
...
```
for x86_64, resulting on cross-compiling vmtest for x86_64

> Might be nice to pull the code into a file in `scripts/` but not a big deal

Agreed, let me do this....

chantra commented:
moved to a script. Fixed some potential issue (e.g do not install toolchain for the running platform).

workflow run: https://github.com/chantra/danobi-vmtest/actions/runs/7479502764/job/20356831774
released artifacts: https://github.com/chantra/danobi-vmtest/releases/tag/v2.0.0

danobi reviewed (approved):
thanks!

danobi commented on scripts/build_release.sh:
```diff
@@ -0,0 +1,48 @@
+#
+# Usage:
+#   ./scripts/build_release.sh
+#   ./scripts/build_kernel.sh x86_64 aarch64
```
typo?
```suggestion
#   ./scripts/build_release.sh x86_64 aarch64
```

chantra replied:
done, thanks

chantra commented on scripts/build_release.sh:
```diff
@@ -0,0 +1,48 @@
+for arch in "${ARCHS[@]}"; do
+    # Compile the binary
+    RUSTFLAGS="-C target-feature=+crt-static -C linker=/usr/bin/${arch}-linux-gnu-gcc" cargo build --release --target "${arch}-unknown-linux-gnu"
+    cp "./target/${arch}-unknown-linux-gnu/release/vmtest" "./vmtest-$(gnu_to_debian "${arch}")"
```
@danobi  any reason to not use $arch as emitted by `uname -m`? Asking mostly because it is easier to know what binary to download in by by doing something like `curl http://....../${release}/vmtest-$(uname -m)` rather than doing the translation dance.

danobi replied:
There was probably a consistency reason a while ago but I forget why. Feel free to change it - it sounds like there's a practical reason to

chantra replied:
ok,  change applied. Example release: https://github.com/chantra/danobi-vmtest/releases/tag/v2.1.0

chantra replied:
side-effect: https://github.com/danobi/vmtest-action/blob/fed44e099e1e7b79251780393453929ab20d107e/vmtest.js#L67 would need to be change when version is bumped.
